services/net/image/ImageAction.cs

```diff
@@ -105,6 +105,45 @@ public ImageAction(IKafkaMessenger kafka, IApiService api, IOptions<ImageOptions
         }
     }
 
+    /// <summary>
+    /// Fetch the image from the remote data source based on configuration.
+    /// </summary>
+    /// <param name="dataSource"></param>
+    /// <returns></returns>
+    /// <exception cref="NotImplementedException"></exception>
+    private async Task FetchImage(DataSourceModel dataSource)
+    {
+        // TODO: use private keys in ./keys folder to connect to remote data source.
+        // TODO: Fetch image from source data location.  Only continue if the image exists.
+        // TODO: Eventually handle different data source locations based on config.
+        throw new NotImplementedException();
+    }
+
+    /// <summary>
+    /// Perform image processing based on configuration.
+    /// </summary>
+    /// <param name="dataSource"></param>
+    /// <returns></returns>
+    /// <exception cref="NotImplementedException"></exception>
+    private async Task ProcessImage(DataSourceModel dataSource)
+    {
+        // TODO: Process Image based on configuration.
+        throw new NotImplementedException();
+    }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="dataSource"></param>
+    /// <returns></returns>
+    /// <exception cref="NotImplementedException"></exception>
+    private async Task CopyImage(DataSourceModel dataSource)
```

Comment:
Not sure if file copy will be async.  If it's not, just change the function accordingly.